picobot/painter.py

```diff
@@ -262,31 +273,34 @@ def sticker_from_text(
     )
     draw_time(dr, text=msg_time, points=points_balloon, font=time_font)
 
-    ratio = 512 / img_width
-    sample = Image.ANTIALIAS
-    img = img.resize((512, int(ratio * size[1])), resample=sample)
+    img = resize_to_sticker_limits(img)
     img_path = IMG_DIR / f'{IMG_PREFIX}{user_id}.png'
     img.save(img_path)
     img.close()
     return img_path
 
 
-def wrapped_text(text: str, line_limit=25):
-    return '\n'.join(textwrap.wrap(text, width=line_limit))
+def wrapped_text(text: str, line_width=25, max_lines=None):
+    return '\n'.join(textwrap.wrap(text, width=line_width, max_lines=max_lines))
 
 
 def sticker_from_image(jpg_path: Path):
     img: Image = Image.open(jpg_path)
+    img = resize_to_sticker_limits(img)
+    img_path = jpg_path.with_suffix('.png')
+    img.save(img_path)
+    img.close()
```

Comment:
We can ensure the `img.close` using an `with` statement:

```suggestion
    with Image.open(jpg_path) as img:
        img = resize_to_sticker_limits(img)
        img_path = jpg_path.with_suffix('.png')
        img.save(img_path)
```